brave/brave-core
# Add additional entry points to Leo acquisition source metric, add most used Leo entry point metric

Resolves https://github.com/brave/brave-browser/issues/40406



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

For `Brave.AIChat.AcquisitionSource`:

0. Start with fresh profile, ensure metric does not exist.
1. Trigger Leo AI using one of the entry points, and send a request to Leo.
2. Ensure metric now exists with the correct value.
3. Repeat steps 0 - 2 for each entry point.

For `Brave.AIChat.MostUsedEntryPoint`

0. Start with fresh profile, ensure metric does not exist.
1. Trigger Leo AI using one of the entry points, and send a request to Leo.
2. Ensure metric exists with the correct value.
3. Close Leo via the X button in the panel.
4. Open Leo via other entry points, ensure metric has correct value indicating the entry point with the most usages. Only close Leo via the X button in the panel before triggering an entry point.
5. Ensure  `Brave.AIChat.AcquisitionSource` stays constant throughout the test.
